src/Inventory.zig

```diff
@@ -1645,8 +1645,12 @@ pub const Command = struct { // MARK: Command
 			if(!self.dest.canHold(.{.item = sourceStack.item, .amount = self.amount})) return;
 
 			var remainingAmount = self.amount;
+			var selectedEmptySlot: i32 = -1;
 			for(self.dest._items, 0..) |*destStack, destSlot| {
-				if(std.meta.eql(destStack.item, sourceStack.item) or destStack.item == null) {
+				if(destStack.item == null and selectedEmptySlot == -1) {
```

Comment:
Use optional, instead of random number. This also likely crashes if there is no empty slot.

Reply:
Done :+1: